core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -237,15 +237,15 @@ Path getMetadataFile(int metadataVersion) throws IOException {
     for (TableMetadataParser.Codec codec : TABLE_METADATA_PARSER_CODEC_VALUES) {
       Path metadataFile = metadataFilePath(metadataVersion, codec);
       FileSystem fs = getFileSystem(metadataFile, conf);
-      if (fs.exists(metadataFile)) {
+      if (fs.isFile(metadataFile)) {
```

Comment:
`isFile` method isn't marked as a deprecated in this version, but [it will be deprecated in newer versions](https://hadoop.apache.org/docs/r3.4.0/api/org/apache/hadoop/fs/FileSystem.html#isFile-org.apache.hadoop.fs.Path-) and the current java doc says:
```
Note: Avoid using this method. Instead reuse the FileStatus 
   * returned by getFileStatus() or listStatus() methods.
```

Reply:
s3a has [specialized implementation](https://github.com/apache/hadoop/blob/b55c9c7495a7da6a50082259ffd143cb8c1ee424/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L5048) of `isFile` that only use headobject. If we switch to `getFileStatus` then this PR become non sense. 
What do you think?

Reply:
Or maybe we use S3FileIO here for metadata file operations? The S3FileIO seems to perform neat file exist check.

Reply:
We definitely shouldn't try to mix use of HadoopFileIO/FileSystem and S3FileIO since they have different behaviors and configuration.  I think the additional call is a product of trying to achieve correct file system semantics on an object store.  Unless they remove the deprecation or provide a separate utility to perform this check, I'm not sure there's a lot of alternatives.

Reply:
I don't mean to use S3FileIO here, but just use the base FileIO interface for read/write/check manifest files, and the FileIO's javadoc says it is capable for this.
The s3a's additional call I guess is for the folder semantics, but if iceberg knows manifest files are "file", the listobject can be eliminated.

Reply:
I've updated them with FileIO. Users who configures a S3FileIO can benefit from it, and the default HadoopFileIO basically behaves the same as the fs.exists.